src/Tribe/Modules/Single_Event_Template.php

```diff
@@ -0,0 +1,169 @@
+		if ( isset( $fieldMap['description'] ) ) {
+			unset( $fieldMap['description'] );
+		}
+		$fieldMap['unsafe_description'] = 'post_content';
+		$fieldMap['start_date_utc']     = 'EventUTCStartDate';
+		$fieldMap['end_date_utc']       = 'EventUTCEndDate';
+
+		return $fieldMap;
+	}
+
+	/**
+	 * Create a block editor version of the imported event(s) based on the draft event defined below by post ID.
+	 *
+	 * @param object $event The event object being handled.
+	 * @param object $data  The import metadata object.
+	 *
+	 * @return object The event object.
+	 */
+	public function tec_ea_single_event_template( object $event, object $data ): object {
+
+		// Make sure this is only done on valid origins.
+		$valid_origins = [
+			'ical',
+			'gcal',
+			'ics',
+		];
+
+		/**
+		 * Allows filtering the accepted origins.
+		 * Useful if you want to limit it to certain type(s) of source(s).
+		 *
+		 * @var array $valid_origins The array of origins.
+		 */
+		$valid_origins = apply_filters( 'tribe_ext_ea_additional_options_valid_origins', $valid_origins );
+
+		if ( ! in_array( $data->meta['origin'], $valid_origins ) ) {
+			return $event;
+		}
+
+		$valid_sources = [];
+
+		/**
+		 * Allows filtering the source URLs to which the template should be applied to.
+		 * If you want to limit this to a certain source URL, set it here.
+		 *
+		 * @var array $valid_sources The array of the import URLs
+		 */
+		$valid_sources = apply_filters( 'tribe_ext_ea_additional_options_valid_sources', $valid_sources );
+
+		if (
+			! empty( $valid_sources )
+			&& ! in_array( $data->meta['source'], $valid_sources )
+		) {
+			return $event;
+		}
+
+		// Get the template from the saved setting.
+		$template_post_id = tribe_get_option( Settings::PREFIX . 'default_template', false );
+
+		if ( $template_post_id == 'enable' ) {
+			$blocks = [
+				'datetime'       => '<!-- wp:tribe/event-datetime /-->',
+				'featured_image' => '<!-- wp:tribe/featured-image /-->',
+				'content_start'  => '<!-- wp:paragraph {"placeholder":"Add Description..."} -->',
+				'content'        => '<p>[tec_ea_content]</p>',
+				'content_end'    => '<!-- /wp:paragraph -->',
+				'organizer'      => '<!-- wp:tribe/event-organizer /-->',
+				'venue'          => '<!-- wp:tribe/event-venue /-->',
+				'sharing'        => '<!-- wp:tribe/event-links /-->',
+				'related'        => '<!-- wp:tribe/related-events /-->',
+				'comments'       => '<!-- wp:post-comments-form /-->',
+			];
+
+			$template = implode( "\n", $blocks );
+		} else {
+			$post = get_post( $template_post_id );
+
+			// If there is no template, or not the right format then skip.
+			if (
+				! $post
+				|| $post->post_type != 'tribe_events'
+				|| $post->post_status != 'draft'
+			) {
+				return $event;
+			}
+
+			// Get the post content template.
+			$template = $post->post_content;
+		}
+
+		// The post content from the source site.
+		$description_from_source = $event['post_content'];
```

Comment:
The `$event` parameter is an `object` (see the method signature), do not use it as an array.
```suggestion
		$description_from_source = $event->post_content;
```

Reply:
That is probably my bad. `$event` should come through as an array from here:
https://github.com/the-events-calendar/the-events-calendar/blob/f3481aa60a8f3860f2199b72b3d4175063161c88/src/Tribe/Aggregator/Record/Abstract.php#L1910-L1915
So need to change the signature.
Right?

Reply:
✔️ Updated parameter types.